Another Pylint PR

theJollySin opened the pull request:
This is a low-priority, low-risk, Pylint PR.

I have no excuse for myself. I've just been trying to wrap my head around the code base and understand what's going on, and I tend to do cleanup as I read. I can quit any time I want, I swear.

This PR fixes these Pylint errors:

* **R1711** useless-return
* **W0612** unused-variable
* **R1714** consider-using-in
* **C0113** correct use of not in

Three other issues are also addressed:

* **py2to3** - removed 'from __future__ print import', as this is Python 3 code and these statements appear to do nothing
* Removing circular imports from several places, though some still remain
* Removing useless Pylint code duplication detection, this appeared to be 100% false positives to me.

theJollySin commented:
As far as I can tell, the test coverage decreased for this PR because I removed lines of code.

So... that's fun.

ntouran commented on armi/runLog.py:
```diff
@@ -383,7 +382,7 @@ def flush():
 
 def prompt(statement, question, *options):
-    """"Prompt the user for some information."""
+    """ "Prompt the user for some information."""
```
still has extra "

ntouran reviewed (approved):
Thanks! This kind of broad sweep is really valuable and appreciated. 
I had a few in-line mini-comments that you can ignore. The only main concern is conflicts in the cross section code that strangely was showing up here in the pr. But just rebasing or merging that in may make those go away.

jakehader commented on armi/physics/neutronics/crossSectionGroupManager.py:
```diff
@@ -1131,12 +1135,6 @@ def updateNuclideTemperatures(self, blockCollectionByXsGroup=None):
     "FluxWeightedAverage": FluxWeightedAverageBlockCollection,
 }
 
-HOMOGENEOUS_BLOCK_COLLECTIONS = {
-    "Median": MedianBlockCollection,
-    "Average": AverageBlockCollection,
-    "FluxWeightedAverage": FluxWeightedAverageBlockCollection,
-}
-
```
Yes i did remove these in #289.

theJollySin replied:
Okay, there are no longer any conflicts in the PR here.

theJollySin commented on armi/reactor/blueprints/tests/test_gridBlueprints.py:
```diff
@@ -266,7 +266,7 @@ def test_construct(self):
         geom = systemLayoutInput.SystemLayoutInput()
         geom.readGeomFromStream(io.StringIO(RTH_GEOM))
-        gridDesign = geom.toGridBlueprints("test_grid")[0]
+        _ = geom.toGridBlueprints("test_grid")[0]
```
Actually, now that I look at this a second time, this test doesn't actually have an `assert` in it.  So, perhaps I will throw the `gridDesign` variable back in and test it.

theJollySin commented on armi/bookkeeping/visualization/xdmf.py:
```diff
@@ -287,7 +287,6 @@ def dumpState(
         node = r.p.timeNode
 
         timeGroupName = database3.getH5GroupName(cycle, node)
-        dbVersion = self._inputDb.version
```
I checked. There are no side effects in the "version" getter.  There is one in the setter though. So it would make a difference if we were going in the other direction.

theJollySin commented:
The PR conflicts have been fixed.

Also, because I removed some lines of code as part of the pylint cleanup, it turns out we lost one line of code coverage.  And that made the code coverage Travis job fail.  So I added a some unit tests to the materials module to make it stop complaining.

ntouran commented:
Thanks. We do often accept PRs that have small reductions in coverage so you don't have to jump through hoops to keep it happy. It's just to help us know when we really added or uncovered a lot of untested code.

ntouran commented:
Hm I'm still slightly concerned about the diff view showing big non-trivial changes to the cross section module related to cce36f29a95b1. This is clearly related to conflict resolution but it makes me slightly worried about the merge. 

If it were me, I'd probably rebase onto `origin/master` and deal with the conflicts there first, then update this PR with that. Then the diff views will be clean.

Thoughts on doing that?

theJollySin commented:
> Hm I'm still slightly concerned about the diff view showing big non-trivial changes to the cross section module...

@ntouran  Yeah, sorry about that. I have rebased the branch so those changes go away. If you look in the "files changed" tab you shouldn't see any of the "crossSection" modules have changes any more.

ntouran commented:
perfect. looks great. thanks.
